packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.encryption_extension.test.ts

```diff
@@ -0,0 +1,687 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  pointInTimeFinderMock,
+  mockGetCurrentTime,
+  mockPreflightCheckForCreate,
+  mockGetSearchDsl,
+} from './repository.test.mock';
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import { SavedObjectsRepository } from './repository';
+import { loggerMock } from '@kbn/logging-mocks';
+
+import { elasticsearchClientMock } from '@kbn/core-elasticsearch-client-server-mocks';
+import { kibanaMigratorMock } from '../mocks';
+import { SavedObjectsSerializer } from '@kbn/core-saved-objects-base-server-internal';
+import {
+  ISavedObjectsEncryptionExtension,
+  SavedObjectsRawDocSource,
+} from '@kbn/core-saved-objects-server';
+import {
+  bulkCreateSuccess,
+  bulkGetSuccess,
+  bulkUpdateSuccess,
+  createDocumentMigrator,
+  createRegistry,
+  createSpySerializer,
+  ENCRYPTED_TYPE,
+  findSuccess,
+  getMockGetResponse,
+  mappings,
+  mockTimestamp,
+  mockTimestampFields,
+  mockVersion,
+  mockVersionProps,
+  MULTI_NAMESPACE_ENCRYPTED_TYPE,
+  TypeIdTuple,
+  updateSuccess,
+} from '../test_helpers/repository.test.common';
+import { savedObjectsExtensionsMock } from '@kbn/core-saved-objects-api-server-mocks';
```

Comment:
In this case we're testing that the repository is calling the extension correctly. How should we do this without the mocks? Or do you mean that we should have local mocks for this?

Reply:
Yea, what's we've been doing in such cases is to duplicate the mocks in a `test_helpers` folders to avoid importing from the mock package. It's not ideal but allow to keep our rule of `X-internal` shouldn't import from `X-mocks`. 

FWIW this was caused by the fact that some mock packages actually need the internal package to function (e.g delegating to a concrete implementation for the mock). But that's not the case for the SO packages, so we may be fine just keeping your changes. I'm just a bit concerned that if at some point we do need the mock package to import from the impl one, we will be stuck. But I guess we could duplicate the mocks in the test_helpers folder when this occurs.

Curious to see what @TinaHeiligers thinks about it.

Reply:
++ to Pierre's comment. While we're not running into issues in this particular case, we need to assume that it's purely coincidental and might very well change over time.
We need to be strict and not use a domain's public mock for internal testing.

For now, we can create duplicate mocks and reference them from a `test_helpers` folder within the package.